frontend/src/Frontend/UI/Transfer.hs

```diff
@@ -461,9 +460,50 @@ lookupKeySets logL networkName nodes chain accounts = do
           putLog logL LevelInfo $ "lookupKeysets failed:" <> tshow e
           pure Nothing
 
-    liftIO $ trigger r
+    resolvedReq <- for initReq $ imapM $ \(AccountName name) details -> do
+            let mref = details ^? _AccountStatus_Exists . accountDetails_guard . _AccountGuard_Other . Pact._GKeySetRef . to (\(Pact.KeySetName name) -> name)
+                mbal = details ^? _AccountStatus_Exists . accountDetails_balance . (to unAccountBalance)
+                updateBal old new = if old == new then old else new
+            fmap (fromMaybe details) $ for (liftA2 (,) mbal mref) $ \(bal,ref) -> do
+              let code = printf "{\"balance\" : (at 'balance (coin.details \"%s\")), \"guard\" : (describe-keyset \"%s\")}" name ref
+              cmd <- simpleLocal Nothing networkName pm (T.pack code)
+              doReqFailover envs (Api.local Api.apiV1Client cmd) >>= \case
+                Left _ -> pure details
+                Right cr -> case Pact._crResult cr of
+                  Pact.PactResult (Right pv) -> do
+                    putLog logL LevelInfo "lookupKeySets on ref: success"
+                    let res = fromMaybe details $ do
+                          guard' <- pv ^? Pact.Types.PactValue._PObject . (to Pact._objectMap) . (at "guard") . _Just . Pact.Types.PactValue._PGuard . (to (fromPactGuard mref))
+                          balance <- pv ^? Pact.Types.PactValue._PObject . (to Pact._objectMap) . (at "balance") . _Just . Pact.Types.PactValue._PLiteral . (to (updateBal bal . _lDecimal))
+                          return $ AccountStatus_Exists $ AccountDetails (AccountBalance balance) guard'
+                    putLog logL LevelInfo $ tshow res
+                    return res
+                  Pact.PactResult (Left e) -> do
+                    putLog logL LevelInfo $ "lookupKeySets failed on keyset-ref lookup:" <> tshow e
+                    pure details
+
+
+    liftIO $ trigger resolvedReq
   pure result
 
+-- no signing of any kind here
```

Comment:
Let's move this into `Frontend.Network` so it will be easier for other people to discover later.